addons/account/models/account_move.py

```diff
@@ -4248,24 +4248,18 @@ def _disable_recursion(self, container, key, default=None, target=True):
         :return: True iff we should just exit the context manager
         """
 
-        disabled = container['records'].env.context.get(key, default) == target
-        previous_values = {}
-        previous_envs = set(self.env.transaction.envs)
-        if not disabled:  # it wasn't disabled yet, disable it now
-            for env in self.env.transaction.envs:
-                previous_values[env] = env.context.get(key, EMPTY)
-                env.context = frozendict({**env.context, key: target})
+        stack = self.env.cr.cache.setdefault('disable_recursion_stack', StackMap())
```

Comment:
Nitpicking, since you use a global cache, prefix with "account".
```suggestion
        stack = self.env.cr.cache.setdefault('account_disable_recursion_stack', StackMap())
```